### DISCUSSION TITLE
multiphase filed and polycrystalline

### DISCUSSION THREAD

**Discussion by Ali-toghraee** [idaholab/moose#16626] [Q&A Modules: Phase field] [answered]
Hi all,

-I have a multiphase  field model (metal, hydride, fluid) which uses Allen-Cahn + Cahn-Hillard + TensorMech. I want to extend that model to a poly crystalline one. I dont have grain growth i.e. my grain boundaries dont move, but my hydride phase will grow. Is there any example that I can use as initial step? 
I have looked at "ACGrGrPoly" but it seems it does not accept any F= total-energy 

- If I reduce the model to a bi-crystaline (two metal grains + hydride+ fluid) , what is the best approach to model that? 4 order parameters for  two metal grains + hydride+ fluid ?

Best,

**Comment by Ali-toghraee** [comment]
Hi again,

Any suggestion/help on this topic?

Thanks,

**Reply by jiangwen84** [reply]
@karim-E-ahmed Could you provide some suggestions?

**Comment by karim-E-ahmed** [comment, marked as answer]
If you have the model, you can directly implement it using Allen-cahn and cahn-hilliard kernels with proving the free energy as a derivative parsed material as function of your 4 or more variables. I also suggest you to look at the grand-potential formulation (see the paper, Grand-potential-based phase-field model for multiple phases, grains, and chemical components) a few examples related to that paper are implemented in moose.

**Reply by Ali-toghraee** [reply]
Thank you for the help Karim. I look into the grand intentional examples.
